### ISSUE TITLE
Feature request: Multi-label Binarizer (k-hot)

### ISSUE THREAD

**Issue by MtDersvan** [tensorflow/tensorflow#7094]
Hi, I feel like an array op that converts between iterable of iterables and a multi-label format would be useful for preprocessing in multi-label tasks. Basically a tensorflow version of a sklearn [MultiLabelBinarizer](http://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MultiLabelBinarizer.html) that can iterate through Tensors.
I would assume a `tf.one_hot()` modification can work.
Thanks.

**Comment by drpngx** [comment]
@ilblackdragon is that something we would welcome as a contribution or implement ourselves?

**Comment by Tosync** [comment]
@drpngx @aselle any updates on adding a tensorflow version of the sklearn MultiLabelBinarizer?.

**Comment by drpngx** [comment]
Is that `reduce_max(one_hot)`?

**Comment by MtDersvan** [comment]
@drpngx Almost, it must handle a tensor rank and shape properly, to work with multi-dimensional tensors as well as with one-dimensional, and not to mix up when its a multi-label vs single-label*batch_size.
E.g. `tf.reduce_max(tf.one_hot(labels, num_classes), proper_axis)` where `labels` can have different shapes and batch_sizes:
```
# Example:
# Single-label:
#   b_s==1: [[l1]]
#   b_s==4: [[l1],[l2],[l3],[l4]]
#       axis=0
# Multi-label:
#   b_s==1: [[l1_1,l1_2]]
#   b_s==4: [[l1_1,l1_2],[l2_1,l2_2,l2_3],[l3_1],[l4_1,l4_2]]
```

**Comment by drpngx** [comment]
Feel free to send a PR that we can review.

**Comment by harshini-gadige** [comment]
Feel free to reopen this issue if there are any contributions. Also PR number can be posted here.
